lib/move_requester.rb

```diff
@@ -0,0 +1,29 @@
+class MoveRequester
+  def initialize(ui)
+    @ui = ui
```

Comment:
I think you can setup the class variables as `attr_reader` and `attr_writer` which means you don't need to use the `@` symbol outside of the constructor.